artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java

```diff
@@ -2409,14 +2410,10 @@ private void deliver() {
       // Either the iterator is empty or the consumer is busy
       int noDelivery = 0;
 
-      int size = 0;
-
-      int endPos = -1;
-
       int handled = 0;
 
       long timeout = System.currentTimeMillis() + DELIVERY_TIMEOUT;
-
+      consumers.reset();
```

Comment:
if any exception would be thrown before it is ok that `reset` won't be called? 
If not, better to wrap the whole logic with ìtry..finally` and `reset`

Reply:
Its fine. We actually only want to reset on a succesful handled nomatch or expired. Or before starting iterating